pkg/component/operator/web/v0/crawl_website.go

```diff
@@ -105,6 +117,10 @@ func (e *execution) CrawlWebsite(input *structpb.Struct) (*structpb.Struct, erro
 
 		link := e.Attr("href")
 
+		if !targetLink(link, inputStruct.Filter) {
```

Comment:
nit: this could be more efficient by compiling the expressions in the filter only once, then using them for all the links in the execution. I don't think this will ever be a bottleneck but I wanted to highlight that regular expressions aren't particularly cheap, computationally speaking.